src/data-card-advanced.tsx

```diff
@@ -0,0 +1,463 @@
+    onPress: () => void;
+    Icon?: React.FC<IconProps>;
+    iconSize?: number;
+    iconColor?: string;
+    iconBackground?: string;
+    iconBackgroundInverse?: string;
+};
+
+type CardActionsGroupProps = {
+    actions: Array<CardAction>;
+    isInverse?: boolean;
+};
+
+const CardActionsGroup = ({actions, isInverse}: CardActionsGroupProps): JSX.Element => {
+    return (
+        <Inline space={0}>
+            {actions.map(
+                (
+                    {
+                        onPress,
+                        label,
+                        Icon,
+                        iconSize = 20,
+                        iconColor = vars.colors.neutralHigh,
+                        iconBackground = styles.cardAction,
+                        iconBackgroundInverse = styles.cardActionInverse,
+                    },
+                    index
+                ) =>
+                    Icon ? (
+                        <IconButton
+                            size={48}
+                            key={index}
+                            onPress={onPress}
+                            aria-label={label}
+                            className={styles.cardActionIconButton}
+                            style={{display: 'flex'}}
+                        >
+                            <div className={isInverse ? iconBackgroundInverse : iconBackground}>
+                                <Icon color={iconColor} size={iconSize} />
+                            </div>
+                        </IconButton>
+                    ) : (
+                        <div key={index} className={styles.cardActionIconButton} />
+                    )
+            )}
+        </Inline>
+    );
+};
+
+const useTopActions = (actions?: Array<CardAction>, onClose?: () => void) => {
+    const {texts} = useTheme();
+    const finalActions = actions ? [...actions] : [];
+
+    if (onClose) {
+        finalActions.push({
+            label: texts.closeButtonLabel,
+            onPress: onClose,
+            Icon: IconCloseRegular,
+        });
+    }
+
+    return finalActions;
+};
+
+type CardContentProps = {
+    headline?: string | RendersNullableElement<typeof Tag>;
+    pretitle?: string;
+    pretitleAs?: string;
+    pretitleLinesMax?: number;
+    title?: string;
+    titleAs?: string;
+    titleLinesMax?: number;
+    subtitle?: string;
+    subtitleLinesMax?: number;
+    description?: string;
+    descriptionLinesMax?: number;
+};
+
+const CardContent: React.FC<CardContentProps> = ({
+    headline,
+    pretitle,
+    pretitleAs = 'h4',
+    pretitleLinesMax,
+    title,
+    titleAs = 'h3',
+    titleLinesMax,
+    subtitle,
+    subtitleLinesMax,
+    description,
+    descriptionLinesMax,
+}) => {
+    const {textPresets} = useTheme();
+    const renderHeadline = () => {
+        if (!headline) {
+            return null;
+        }
+        if (typeof headline === 'string') {
+            return <Tag type="promo">{headline}</Tag>;
+        }
```

Comment:
do we want to support this use case? I think you just copy&pasted this from card.tsx, but this is this way in other cards just to support a legacy use case where headline could be just a string

Reply:
could you please elaborate?